### PR TITLE
Fix DATE function for negative parameters

### PR DESCRIPTION
**Pull request by jhkcia**
On excel negative parameters of DATE function works and try to subtract the number from the date, now it matches excel.

### CODE REVIEW

File: test/date-time.js [hunk 1]
```diff
@@ -8,9 +8,21 @@ describe('Date & Time', function () {
     var date = dateTime.DATE(1900, 1, 1);
     date.getFullYear().should.equal(1900);
     date.getMonth().should.equal(1 - 1);
-    date.getDay().should.equal(1);
+    date.getDate().should.equal(1);
 
-    dateTime.DATE(1900, 1, -1).should.equal(error.num);
+    date = dateTime.DATE(1900, 1, -1);
+    date.getFullYear().should.equal(1899);
```
Comment (nicolashefti) [inline test/date-time.js:14]: I agree with this one, but with 1900, I get an error num. Just another mistery of Excel.

```
=DATE(1910; 1;-1) => 1909.12.30
=DATE(1900; 1;-1) => #NUM
```

File: test/date-time.js [hunk 2]
```diff
@@ -8,9 +8,21 @@ describe('Date & Time', function () {
...
+    date.getDate().should.equal(30);
+
+    date = dateTime.DATE(2000, -36, 1);
+    date.getFullYear().should.equal(1996);
+    date.getMonth().should.equal(12 - 1);
+    date.getDate().should.equal(1);
+
+    dateTime.DATE(10, 1, 1).getFullYear().should.equal(1910);
+    dateTime.DATE(-1, 1, 1).should.equal(error.num);
+    // dateTime.DATE(1899, 1, 1).getFullYear().should.equal(3799);
```
Comment (nicolashefti) [inline test/date-time.js:25]: Another weird thing of Excel?

File: test/date-time.js [hunk 3]
```diff
@@ -8,9 +8,21 @@ describe('Date & Time', function () {
...
 
-    dateTime.DATE(1900, 1, -1).should.equal(error.num);
+    date = dateTime.DATE(1900, 1, -1);
+    date.getFullYear().should.equal(1899);
+    date.getMonth().should.equal(12 - 1);
+    date.getDate().should.equal(30);
+
+    date = dateTime.DATE(2000, -36, 1);
+    date.getFullYear().should.equal(1996);
+    date.getMonth().should.equal(12 - 1);
```
Comment (nicolashefti) [inline test/date-time.js:20]: 11 ? Why 12 - 1 ?

File: test/date-time.js [hunk 4]
```diff
@@ -8,9 +8,21 @@ describe('Date & Time', function () {
     var date = dateTime.DATE(1900, 1, 1);
     date.getFullYear().should.equal(1900);
     date.getMonth().should.equal(1 - 1);
-    date.getDay().should.equal(1);
```
Comment (nicolashefti) [inline test/date-time.js:11]: Nicely spoted.

**Review by nicolashefti — COMMENTED** [review]
Thanks for the fix and upgrade. I just have little questions.

**Comment by jhkcia** [comment]
@nicolashefti 
Excel minimum date is 1900/01/01, it does not support any date below that, so `=DATE(1900; 1;-1)`  is an invalid formula and if the year is lower than 1900 it tries to add it to 1900 so DateTime.DATE(1899, 1, 1).getFullYear().should.equal(3799)  is excel behaviour.

`11 ? Why 12 - 1 ?`
in JS month start from zero but in excel it starts from 1. so for assertions, I reduce 1 from the month number.

`date.getDay().should.equal(1);`
getDay return day of the week, I replaced it with getDate which returns the day number of a date.

File: test/date-time.js (hunk 2 above)
Comment (nicolashefti) [reply test/date-time.js:25]: Same as in the other PR. Can you create a xit statement for this one ?
```
xit(‘should be Excel behaviour, but we do not want to recreate it’, function(){
  ...
}
```

**Review by nicolashefti — CHANGES_REQUESTED** [review]

**Review by nicolashefti — APPROVED** [review]

**Pull request merged**